mlr-org/gallery/technical/2025-01-12-efs/index.qmd

```diff
@@ -0,0 +1,417 @@
+  scale_fill_brewer(palette = "Set1") +
+  theme_minimal(base_size = 14)
+```
+
+Alternatively, the **Nogueira** stability measure can be used, which unlike the Jaccard Index, it's a chance-corrected similarity measure [@Nogueira2018]:
+```{r efs-023}
+#| warning: false
+autoplot(efs, type = "stability", stability_measure = "nogueira", 
+         stability_args = list(p = task$n_features)) +
+  scale_fill_brewer(palette = "Set1") +
+  theme_minimal(base_size = 14)
+```
+
+## Feature Ranking
+
+Using the Pareto method, we demonstrated how we can identify a reasonable cutoff for the number of selected features.
+Now we will focus on how to create a consensus ranked feature list based on the results of the ensemble feature selection.
+
+The most straightforward ranking is obtained by counting how often each feature appears in the 'best' feature subsets (`"features"`).
+Below we show the top 8 features, i.e. up to the cutoff derived from the knee point of the estimated Pareto front.
+The column `"score"` represents these counts, while the column `"norm_score"` is the **feature selection frequency** or also known as **selection probability** [@Meinshausen2010]:
+```{r efs-024}
+efs$feature_ranking(method = "av", use_weights = FALSE, committee_size = 8)
+```
+
+In the language of Voting Theory, we call the method that generates these counts *approval voting* (`method = "av"`) [@Lackner2023].
+Using this framework, learners act as *voters*, features act as *candidates* and voters select certain candidates (features).
+The primary objective is to compile these selections into a consensus ranked list of features (a committee).
+The `committee_size` specifies how many (top-ranked) features to return.
+
+Internally, `$feature_ranking()` uses the [`fastVoteR`](https://bblodfon.github.io/fastVoteR/) R package, which supports more advanced ranking methods.
+For example, we can perform **weighted ranking**, by considering the varying performance (accuracy) of each learner.
+This results in the same top 8 features but with slightly different ordering:
+```{r efs-025}
+efs$feature_ranking(method = "av", use_weights = TRUE, committee_size = 8)
+```
+
+Additionally, alternative ranking methods are supported.
+Below, we use *satisfaction approval voting* (SAV), which results in a different feature ranking (e.g. `"V10"` feature is now included in the top 8 features but not `"V4"`):
+```{r efs-026}
+efs$feature_ranking(method = "sav", committee_size = 8)
+```
+
+# EFS-based Feature Selection
+
+The ultimate goal of the ensemble feature selection process is to identify predictive and stable features.
+By combining the ranked feature list with the Pareto-derived cutoff, we can select the final set of features and subset the original dataset for further modeling:
+```{r efs-027}
+n_features = efs$knee_points(type = "estimated")$n_features
+res = efs$feature_ranking(method = "sav", committee_size = n_features)
+task$select(cols = res$feature)
+task
```

Comment:
Do we also want a final model?

Reply:
There is no final model in the efs! Used only for feature selection with any model afterwards you want